sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/serializer/TypeReference.java

```diff
@@ -16,6 +19,8 @@
     private static final ClientLogger LOGGER = new ClientLogger(TypeReference.class);
     private static final String MISSING_TYPE = "Type constructed without type information.";
 
+    private static final Map<Class<?>, TypeReference<?>> CACHE = new ConcurrentHashMap<>();
```

Comment:
Several questions here:
1. Do we only cache the non-Parameterized class? (e.g. Hotel) What about the Parameterized type class like Map<>?
2. Is the Class<?> able to compare in Map?
3. Do we waste memory by using map cache? Search user might have many different models(hotel, book etc). It might improve a bit in time but you cannot predict how large is the cache map.

E.g. if customer just pass you inner subclass, then it is different classes every time make the call. I am afraid it will crash the system since we do not have LRU clean up as normal cache.

Reply:
1. Can only cache non-parameterized types as `ParameterizedType` isn't capable of maintaining `T`, so it will lose generic functionality in return.

2. `Class<?>` could be `Map` but is cannot be `Map<String, Object>` due to #1.

3. The size of this map will be trivial, it would take thousands and thousands of entries for it to register moderate memory usage. It'll help reduce GC pressure when creating a new short-lived `TypeReference` instance for `Class` based instances.